### PR TITLE
feat(atomic): allow user to customize grid card click behavior/links

### LINKED ISSUE #4267: [Bug]: `AtomicResultList` with `display="grid"` can result in a link unexpectedly appearing in newly rendered results
### Which product are you using?

Atomic

### product version

@coveo/atomic-react@2.13.2, @coveo/headless@2.76.0

### bug description

The first result added to the page after clicking load-more-results unexpectedly has a link about its title. This link disappears after clicking anywhere on the page.

https://github.com/user-attachments/assets/e7258821-256a-4f3c-90bb-39d79c0a2be1

### Steps to reproduce

I've been able to reproduce this bug with a minimal implementation of Atomic React in a Create React App.

1. Clone repo: https://github.com/Martin-Alexander/coveo-atomic-issue
2. Run `npm i`
3. Run `npm run start` 
4. Go to http://localhost:3000/
5. Click on load-more-results
6. See that the first newly loaded result unexpectedly has a linked about the title

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by louis-bompart**
Fixes #4267 
https://coveord.atlassian.net/browse/KIT-3470

### CODE REVIEW

File: packages/atomic/src/components/common/item-list/display-grid.tsx [hunk 1]
```diff
@@ -12,35 +11,22 @@ export interface DisplayGridProps {
...
       onClick={(event) => {
         event.preventDefault();
-        select();
-        window.open(
-          item.clickUri,
-          event.ctrlKey || event.metaKey ? '_blank' : gridTarget,
-          'noopener'
-        );
+        ref?.querySelector('atomic-result')?.click();
       }}
```
Comment (louis-bompart) [inline packages/atomic/src/components/common/item-list/display-grid.tsx:28]: Almost useless: but if you put some fat margin/padding, and click in the space between the border of the div and the actual result component, well, your click fell thru.

File: packages/atomic/src/components/search/atomic-result/atomic-result.tsx [hunk 2]
```diff
@@ -149,6 +167,12 @@ export class AtomicResult {
       .join('');
   }
 
+  private getLinkHTML() {
+    return Array.from(this.linkContent!.children)
+      .map((child) => child.outerHTML)
+      .join('');
+  }
```
Comment (louis-bompart) [inline packages/atomic/src/components/search/atomic-result/atomic-result.tsx:174]: - [ ] will prolly refactor into an util 'documentfragment2string'

**Comment by louis-bompart** [comment]
- [ ] add support for other item-template/item-link/*-list

**Review by pathiery — APPROVED** [review]
LGTM for atomic file

File: packages/headless/src/features/commerce/product-templates/product-templates-manager.ts [hunk 3]
```diff
@@ -21,6 +24,7 @@ export interface ProductTemplatesManager<Content = unknown> {
    * @returns (Content) The content of the selected template, or null if no template can be selected for the given product.
    */
   selectTemplate: (product: Product) => Content | null;
+  selectLinkTemplate: (product: Product) => LinkContent | null;
```
Comment (Spuffynism) [inline packages/headless/src/features/commerce/product-templates/product-templates-manager.ts:27]: I think we're missing doc for this

**Review by Spuffynism — APPROVED** [review]
lgtm for the headless commerce part

File: packages/headless/src/features/commerce/product-templates/product-templates-manager.ts (hunk 3 above)
Comment (louis-bompart) [reply packages/headless/src/features/commerce/product-templates/product-templates-manager.ts:27]: Good point, yep, will do.
- [ ] JSDOC

**Review by olamothe — APPROVED** [review]

**Pull request merged**